pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -338,6 +338,10 @@ self: super: {
     dependencies = with self; [ nvim-yarp ];
   });
 
+  denops-vim = super.denops-vim.overrideAttrs (old: {
+    dependencies = with self; [ deno ];
```

Comment:
I don't think this works, see https://github.com/NixOS/nixpkgs/pull/207195#discussion_r1055830519

Reply:
Ah, I missed that (closed but) pre-existing PR. Could it be that Deno isn't on `$PATH`?

Reply:
Yes, and adding it to `dependencies` wouldn't add it to `PATH`

it was also mentioned in the original pr that `it tries to write in the nix store`, can you check whether that is still the case?

Reply:
> Yes, and adding it to dependencies wouldn't add it to PATH

Is there an easy way to set an option from a package? We can set Deno's `$PATH` by setting `g:denops#deno` to a full path.

Alternatively, is there a way to ensure that a package's binaries are on `$PATH` for a plugin?

Reply:
Hmm, I've found a way by using `sed` to patch in the `g:denops#deno` variable. It's pretty ugly though:
```sh
sed -i "1s,^,let g:denops#deno = '\${deno}/bin/deno'\n," \
-i "$out"/plugin/denops.vim
```

I haven't tested this, but I think it should work.

Reply:
> We can set Deno's location by setting g:denops#deno to a full path.

There is an open issue for this, but right now it would require manual configuration on the user's end: https://github.com/NixOS/nixpkgs/issues/172538

> Alternatively, is there a way to ensure that a package's binaries are on $PATH for a plugin?

There is an open pull request to try to implement this: https://github.com/NixOS/nixpkgs/pull/208029

Reply:
@figsoda opinions on the patch? Or should we wait for https://github.com/NixOS/nixpkgs/pull/208029 to get merged and use that?

Reply:
might be better to patch this line instead: https://github.com/vim-denops/denops.vim/blob/44baa0666e69976dd82311a67a220c7f71273368/autoload/denops.vim#L21, note that we usually prefer `substituteInPlace` for the simplicity, but you can use whatever you want to use

we can also just not patch it and require the user to install deno themselves

> it was also mentioned in the original pr that it tries to write in the nix store, can you check whether that is still the case?

this is what I'm more worried about